src/dymaptic.GeoBlazor.Core/Components/Layers/KMLLayer.cs

```diff
@@ -0,0 +1,58 @@
+﻿using dymaptic.GeoBlazor.Core.Exceptions;
+using Microsoft.AspNetCore.Components;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace dymaptic.GeoBlazor.Core.Components.Layers;
+
+/// <summary>
+///     The KMLLayer class is used to create a layer based on a KML file (.kml, .kmz). KML is an XML-based file format used
+///     to represent geographic features.
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-KMLLayer.html">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+public class KMLLayer : Layer
+{
+    /// <summary>
+    ///     Parameterless constructor for use as a razor component
+    /// </summary>
+    public KMLLayer()
+    {
+    }
+    /// <summary>
+    ///     Constructor for use in code
+    /// </summary>
+    /// <param name="url">
+    ///     The url for the KML Layer source data.
+    /// </param>
+    public KMLLayer (string? url = null, PortalItem? portalItem = null)
+    {
+        if (url is null && portalItem is null)
+        {
+            throw new MissingRequiredOptionsChildElementException(nameof(KMLLayer),
+                new[] { nameof(Url), nameof(PortalItem) });
+        }
+#pragma warning disable BL0005
+        Url = url;
+#pragma warning restore BL0005
+    }
+
+    /// <inheritdoc />
+    [JsonPropertyName("type")]
+    public override string LayerType => "kml";
+
+    /// <summary>
+    ///     The url for the KML Layer source data.
+    /// </summary>
+    [Parameter]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RequiredProperty(nameof(PortalItem))]
+    public string? Url { get; set; }
+
```

Comment:
OK, so your code here and in the TypeScript says you are supporting `PortalItem`, but you haven't implemented it.

You need
1. A new `public PortalItem? PortalItem { get; set; }`. _NOT_ a `[Parameter]`
2. Implement `protected override async Task RegisterChildComponent(MapComponent child)`
3. Implement `protected override async Task UnRegisterChildComponent`
4. Implement `internal void ValidateRequiredChildred()`.

This is how nested markup components get registered to work. You could create a test in `Core.Test.Blazor` or a separate sample that lays out a KML like this
```html
<KmlLayer>
    <PortalItem Id="..." />
</KmlLayer>
```

Reply:
refactored the main class and adjusted the markup component to use the url parameter. Also created a layer test class for the kml layer and successfully tested.  Thanks for the assist @TimPurdum   Additional layers can also be added in the future to this test file.